src/pages/BrowseBook/BrowseDocumentModel.js

```diff
@@ -0,0 +1,122 @@
+                    .map(s => [s.split('/')[3], s.split('/')[5]]);
+                const state = {newSearchString: data.payload};
+                attScopes.forEach(s => state[s[0] = s[1]])
+            renderer.config.blockStack.push(
+                    <Link
+                        to={{
+                            pathname: "/search/text",
+                            state
+                        }}
+                        key={renderer.config.nextKey++}
+                        className="browserWord"
+                    >
+                        {data.payload}
+                    </Link>
+            );
+            } else {
+                renderer.config.blockStack.push(data.payload);
+            }
+        }
+    );
+    dInstance.addAction(
+        'scope',
+        () => true,
+        (renderer, context, data) => {
+            const scopeName = data.payload.split('/')[0];
+            const scopeValue = data.payload.split('/').slice(1).join('/');
+            if (data.subType === 'start' && scopeName === 'verses') {
+                renderer.config.currentVerses = scopeValue;
+                const currentChapter = renderer.config.currentChapter;
+                let verseRef = null;
+                let className = "verses_label";
+                if (
+                    renderer.config.currentChapter === renderer.config.selectedChapter &&
+                    renderer.config.currentVerses === renderer.config.selectedVerses
+                ) {
+                    verseRef = renderer.config.selectedVerseRef;
+                    className = "selected_verses_label";
+                }
+                renderer.config.blockStack.push(
+                    <span
+                        id={`v_${renderer.config.currentChapter}_${scopeValue}`}
+                        onClick={() => renderer.config.versesCallback(currentChapter, scopeValue)}
+                        ref={verseRef}
+                        key={renderer.config.nextKey++}
+                        className={className}
+                    >
+                        {scopeValue}
+                    </span>);
+            } else if (data.subType === 'end' && scopeName === 'verses') {
+                renderer.config.currentVerses = null;
+            } else if (data.subType === 'start' && scopeName === 'chapter') {
+                renderer.config.currentChapter = scopeValue;
+                renderer.config.blockStack.push(
+                    <span
+                        key={renderer.config.nextKey++}
+                        className="chapter_label"
+                    >
+                        {scopeValue}
+                    </span>);
+            }
+        }
+    );
+    dInstance.addAction(
+        'endBlock',
+        () => true,
+        (renderer, context) => {
+            const pRef = renderer.config.rendered.length === 0 ? renderer.config.topDocRef : null;
```

Comment:
Here pRef is null in all cases, again not sure of what it would be used for or how topDocRef is set but for Gal 5 it's null everytime